homeassistant/components/sensor/vasttrafik.py

```diff
@@ -0,0 +1,118 @@
+"""
+Support for Västtrafik public transport.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.vasttrafik/
+"""
+from datetime import datetime
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['vtjp==0.1.11']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DELAY = 'delay'
+CONF_DEPARTURES = 'departures'
+CONF_FROM = 'from'
+CONF_HEADING = 'heading'
+CONF_KEY = 'key'
+CONF_NAME = 'name'
+CONF_SECRET = 'secret'
+
+ICON = 'mdi:train'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+PLANNER = None
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_KEY): cv.string,
+    vol.Required(CONF_SECRET): cv.string,
+    vol.Optional(CONF_DEPARTURES): [{
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_DELAY, default=0): cv.positive_int,
+        vol.Optional(CONF_HEADING): cv.string,
+        vol.Optional(CONF_NAME): cv.string}]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the departure sensor."""
+    from vasttrafik import JournyPlanner
+    global PLANNER
```

Comment:
You can avoid the GLOBAL by passing it into the `VasttrafikDepartureSensor`constructor.

Less globals = 👍